core/src/main/java/feign/ResponseInterceptor.java

```diff
@@ -27,12 +27,11 @@ public interface ResponseInterceptor {
   ResponseInterceptor DEFAULT = InvocationContext::proceed;
 
   /**
-   * Called for response around decode, must either manually invoke
-   * {@link InvocationContext#proceed} or manually create a new response object
+   * Called by {@link ResponseHandler} after refreshing the response and wrapped around the whole decode process,
+   * must either manually invoke {@link InvocationContext#proceed} or manually create a new response object
    *
-   * @param invocationContext information surrounding the response been decoded
+   * @param invocationContext information surrounding the response being decoded
    * @return decoded response
    */
-  Object aroundDecode(InvocationContext invocationContext) throws IOException;
```

Comment:
This rename is a no go, as breaks existing api for anyone that implemented a ResponseInterceptor